### PR TITLE
Replaced usages of the deprecated HashDict module with Map in sequence.ex

### PR DESCRIPTION
**Pull request by cryo28**

### CODE REVIEW

File: lib/ex_machina/sequence.ex [hunk 1]
```diff
@@ -1,6 +1,6 @@
 defmodule ExMachina.Sequence do
   def start_link do
-    Agent.start_link(fn -> HashDict.new end, name: __MODULE__)
+    Agent.start_link(fn -> %{} end, name: __MODULE__)
```
Comment (pdawczak) [inline lib/ex_machina/sequence.ex]: How about using [`Map.new/0`](http://elixir-lang.org/docs/stable/elixir/Map.html#new/0) to keep similar level of abstraction?

File: lib/ex_machina/sequence.ex [hunk 2]
```diff
@@ -23,7 +23,7 @@ defmodule ExMachina.Sequence do
      end
   """
   def reset do
-    Agent.update(__MODULE__, fn(_) -> HashDict.new end)
+    Agent.update(__MODULE__, fn(_) -> %{} end)
```
Comment (pdawczak) [inline lib/ex_machina/sequence.ex]: Same as comment above

**Comment by paulcsmith** [comment]
Thank you! I think this is a great change. I like the idea @pdawczak had to use `Map.new`. Once that's switched over I'll go ahead and merge this in!

**Comment by cryo28** [comment]
@paulcsmith Sure 13ecb7c8d317fbed010ef00b31370b4523cd46b1

**Comment by paulcsmith** [comment]
Thank you very much @cryo28!

**Pull request merged**